### PR TITLE
restore disabled testSwiftRunSIGINT test

### PR DESCRIPTION
**Pull request by tomerd**
motivation: test coverage

changes: improve state machine to handle the case the output is fully buffered and enable the test

rdar://91024625

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci smoke test

File: Tests/FunctionalTests/MiscellaneousTests.swift [hunk 1]
```diff
@@ -440,23 +437,27 @@ class MiscellaneousTestCase: XCTestCase {
                 guard let output = String(bytes: bytes, encoding: .utf8) else {
                     return
                 }
+                print(output, terminator: "")
                 self.lock.withLock {
                     switch self.state {
                     case .idle:
-                        self.state = .buffering(output)
+                        self.state = processOutput(output)
```
Comment (abertelrud) [inline Tests/FunctionalTests/MiscellaneousTests.swift:444]: Just curious if this could also have been done with a `fallthrough` here?

Comment (abertelrud) [reply Tests/FunctionalTests/MiscellaneousTests.swift:444]: The separate function seems cleaner, arguably, mostly curious if I understand the logic here correctly.

**Review by abertelrud — APPROVED** [review]

File: Tests/FunctionalTests/MiscellaneousTests.swift (hunk 1 above)
Comment (tomerd) [reply Tests/FunctionalTests/MiscellaneousTests.swift:444]: cannot use `fallthrough ` afaik since we need to bind the variable

**Pull request merged**